### PR TITLE
Show 404 page on not found categories

### PR DESCRIPTION
**Pull request by hlohaus**
### 1. Why is this change necessary?

To show correct error page instead a redirect to home.

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by pantrtxp** [comment]
Hi @hlohaus,

I've checked out your branch and do not get to a 404 page in production mode if I open a Category which does not exist

can you please check this behavior or explain how I can reproduce your workflow?

Sunny regads
Pantrtxp

**Comment by hlohaus** [comment]
Try open /navigation/09ec6e37ee754b74abcdc95601baffff

**Comment by pantrtxp** [comment]
Sadly, I can not confirm this behavior. 
If i check out your branch, set it up in production mode, and go to /navigation/someFakeUUID i'll get an `Category "88bae2d5fde54278a8211da8c9c27f2d" not found` exception in `platform/src/Core/Content/Category/Service/NavigationLoader.php`

**Comment by pantrtxp** [comment]
Hi @hlohaus 

Sorry about the delay.
If you rebase your branch, I'll look into this again.

Sunny regards
Pantrtxp

**Comment by pantrtxp** [comment]
@hlohaus we have some failing phpunit tests
`StorefrontRoutingTest::testForwardFromAddPromotionToApiFails` and 
`ProductExportControllerTest::testInvalidData`

could you please take a look?

**Comment by pantrtxp** [comment]
Hi @hlohaus 

```
Failed asserting that 412 is identical to 404.
 /app/platform/src/Storefront/Test/Controller/StorefrontRoutingTest.php:47
```

This Test is stell failing
Can you look this up please?

**Comment by hlohaus** [comment]
Yes, thats right. InvalidRouteScopeException has 412 as status code

File: src/Storefront/Test/Framework/Controller/ProductExportControllerTest.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function testInvalidData(): void
         $client = $this->createSalesChannelBrowser(null, true);
         $client->request('GET', getenv('APP_URL') . '/export/foo/bar');
 
-        static::assertEquals(500, $client->getResponse()->getStatusCode());
+        static::assertEquals(404, $client->getResponse()->getStatusCode());
```
Comment (pantrtxp) [inline src/Storefront/Test/Framework/Controller/ProductExportControllerTest.php]: Please us the constant here as well

**Review by pantrtxp — CHANGES_REQUESTED** [review]

**Comment by pantrtxp** [comment]
Thank you for your contribution @hlohaus

**Pull request closed without merging**